hub/src/server/http.ts

```diff
@@ -74,7 +77,18 @@ export function makeHttpServer(config: HubConfigInterface): { app: express.Appli
         } else {
           writeResponse(res, { message: 'Server Error' }, 500)
         }
-      })
+      }
+    }
+
+    if (!asyncMutex.tryAcquire(endpoint, handleRequest)) {
```

Comment:
I notice that we _aren't_ guarding `list-files`, `revoke-all`, etc. with a per-path mutex.  Do we want to be doing this?

Reply:
My feeling is no. `list-files` is readonly and should never be to the same path as a resource endpoint. Not sure about `revoke-all`, I don't think it would be used frequently enough to cause concurrency issues but perhaps possible. What do you think?

Reply:
I think it's fine for now, since the authenticator is the only client that will call this method.  We can fix it later if it proves to be a problem.